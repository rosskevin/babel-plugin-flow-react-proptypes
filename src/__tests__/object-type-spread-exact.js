const babel = require('babel-core');
const content = `
type A = {
    foo: string
} 

type Props = {
    bar: string,
    ...$Exact<A>
};

class MyComponent extends React.Component {
    props: Props;
}
`;

it('object-type-spread-exact', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
