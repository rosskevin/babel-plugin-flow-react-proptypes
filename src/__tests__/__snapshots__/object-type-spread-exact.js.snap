// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`object-type-spread-exact 1`] = `
"\\"use strict\\";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\"); } return call && (typeof call === \\"object\\" || typeof call === \\"function\\") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== \\"function\\" && superClass !== null) { throw new TypeError(\\"Super expression must either be null or a function, not \\" + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var MyComponent = function (_React$Component) {
    _inherits(MyComponent, _React$Component);

    function MyComponent() {
        _classCallCheck(this, MyComponent);

        return _possibleConstructorReturn(this, (MyComponent.__proto__ || Object.getPrototypeOf(MyComponent)).apply(this, arguments));
    }

    return MyComponent;
}(React.Component);

MyComponent.propTypes = {
    bar: require(\\"prop-types\\").string.isRequired,
    foo: require(\\"prop-types\\").string.isRequired
};"
`;
